src/api/storage-buckets.tsx

```diff
@@ -0,0 +1,115 @@
+import { handleResponse } from "util/helpers";
+import type { LxdStorageBucket } from "types/storage";
+import type { LxdApiResponse } from "types/apiResponse";
+import { withEntitlementsQuery } from "util/entitlements/api";
+import { fetchStoragePools } from "./storage-pools";
+import { continueOrFinish, pushFailure, pushSuccess } from "util/promises";
+import type { LxdOperationResponse } from "types/operation";
+import { cephObject } from "util/storageOptions";
+
+export const storageBucketEntitlements = ["can_delete", "can_edit"];
+
+const bucketCompatibleDrivers = [cephObject];
+export const isBucketCompatibleDriver = (driver: string): boolean => {
+  return bucketCompatibleDrivers.includes(driver);
+};
+
+export const fetchStorageBucketsFromPool = async (
+  pool: string,
+  isFineGrained: boolean | null,
+  project: string,
+): Promise<LxdStorageBucket[]> => {
+  const entitlements = withEntitlementsQuery(
+    isFineGrained,
+    storageBucketEntitlements,
+  );
+
+  return new Promise((resolve, reject) => {
+    fetch(
+      `/1.0/storage-pools/${pool}/buckets?project=${project}&recursion=1${entitlements}`,
+    )
+      .then(handleResponse)
+      .then((data: LxdApiResponse<LxdStorageBucket[]>) => {
+        resolve(data.metadata.map((bucket) => ({ ...bucket, pool })));
+      })
+      .catch(reject);
+  });
```

Comment:
```suggestion
  return fetch(
    `/1.0/storage-pools/${pool}/buckets?project=${project}&recursion=1${entitlements}`,
  )
    .then(handleResponse)
    .then((data: LxdApiResponse<LxdStorageBucket[]>) => {
      return data.metadata.map((bucket) => ({ ...bucket, pool }));
    });
```
slightly simplified.